File: Daily Test/Python/Bishop Pairs - Attack 4.11.2025.py
n = int(input())
board = [input().split() for _ in range(n)]
res = 0
diag = [1, 1]
rev = [1, -1]
for i in range(n):
    for j in range(n):
        if board[i][j] == '1':
            nc, nr = i + diag[0], j + diag[1]
            while 0 <= nc < n and 0 <= nr < n:
                if board[nc][nr] == '1':
                    res += 1
                nc += 1
                nr += 1
            ec, er = i + rev[0], j + rev[1]
            while 0 <= ec < n and 0 <= er < n:
                if board[ec][er] == '1':
                    res += 1
                ec += 1
                er -= 1
print(res)

File: Daily Test/Java/Bishop Pairs - Attack 4.11.2025.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        String[][] board = new String[n][n];
        for(int i=0; i<n; i++)
            for(int j=0; j<n; j++)
                board[i][j] = sc.next();
        int res = 0;
        int[] diag = {1, 1};
        int[] rev = {1, -1};
        for(int i=0; i<n; i++) {
            for(int j=0; j<n; j++) {
                if(board[i][j].equals("1")) {
                    int nc = i + diag[0], nr = j + diag[1];
                    while(nc >= 0 && nc < n && nr >= 0 && nr < n) {
                        if(board[nc][nr].equals("1")) res++;
                        nc++; nr++;
                    }
                    int ec = i + rev[0], er = j + rev[1];
                    while(ec >= 0 && ec < n && er >= 0 && er < n) {
                        if(board[ec][er].equals("1")) res++;
                        ec++; er--;
                    }
                }
            }
        }
        System.out.println(res);
        sc.close();
    }
}